src/IECoreScene/MeshAlgoDistributePoints.cpp

```diff
@@ -134,39 +139,93 @@ T triangleInterpolatedPrimVarValue(
+	MeshPrimitivePtr &processedMesh,
+	PrimitiveVariable &uvVar, PrimitiveVariable &densityVar,
+	PrimitiveVariable &faceAreaVar, PrimitiveVariable &textureAreaVar
 )
 {
 	if( !mesh )
 	{
 		throw InvalidArgumentException( "MeshAlgo::distributePoints : The input mesh is not valid" );
 	}
 
-	// \todo - remove unused primvars
-	triangulatedMesh = MeshAlgo::triangulate( mesh, canceller );
+	IECoreScene::MeshPrimitivePtr meshWithUsedPrimVars = new MeshPrimitive();
+
+	// We need the topolgy of the source mesh to triangulate it
+	meshWithUsedPrimVars->setTopologyUnchecked( mesh->verticesPerFace(), mesh->vertexIds(), mesh->variableSize( PrimitiveVariable::Vertex ), mesh->interpolation() );
+
+	// Note that we do not transfer creases or corners - they do not affect distribution of points. If we were
+	// to add support for distributing onto the limit surface of a subdiv, then we might need to keep those ...
+	// but that would need to happen on an untriangulated mesh anyway.
+
+	// Transfer the subset of variables that we need
+	for( auto var : mesh->variables )
+	{
+		if( var.first == position )
+		{
+			// \todo - is it a problem that this P that we are setting up solely to be passed in
+			// triangulate ( where it isn't used ), may not have a Point interpretation set correctly?
+			// We don't want to set it yet because the triangulation of this may be driving both P and
+			// the original name given by "position" if it is listed in primitiveVariables, and that
+			// should have the original interpretation.
+			meshWithUsedPrimVars->variables[ "P" ] = var.second;
+			continue;
+		}
+
+		if( var.first == uvSet || var.first == densityMask || var.first == position )
+		{
+			meshWithUsedPrimVars->variables[ var.first ] = var.second;
+			continue;
+		}
+
+		if( var.first == "P" )
+		{
+			// If P wasn't already transferred above, it is being overwritten by whatever does match `position`
+			continue;
+		}
+
+		if( StringAlgo::match( var.first, primitiveVariables ) )
+		{
+			meshWithUsedPrimVars->variables[ var.first ] = var.second;
+		}
+	}
 
-	if ( !triangulatedMesh || !triangulatedMesh->arePrimitiveVariablesValid() )
+	processedMesh = MeshAlgo::triangulate( meshWithUsedPrimVars.get(), canceller );
+	if ( !processedMesh || !processedMesh->arePrimitiveVariablesValid() )
 	{
 		throw InvalidArgumentException( "MeshAlgo::distributePoints : The input mesh could not be triangulated" );
 	}
 
-	auto positionIt = triangulatedMesh->variables.find( position );
+	auto pIt = processedMesh->variables.find( "P" );
 	if(
-		positionIt == triangulatedMesh->variables.end() ||
-		positionIt->second.interpolation != PrimitiveVariable::Vertex ||
-		positionIt->second.data->typeId() != V3fVectorDataTypeId
+		pIt != processedMesh->variables.end() &&
+		pIt->second.interpolation == PrimitiveVariable::Vertex &&
+		pIt->second.data->typeId() == V3fVectorDataTypeId
 	)
+	{
+		processedMesh->variables["P"] = pIt->second;
```

Comment:
What is this assignment for? Since `pIt == find( "P" )` then aren't we just assigning the same data again?